test/functional.spec.js

```diff
@@ -57,43 +65,80 @@ describe('Basic Functionality', function () {
-      const res = await fetch(`http://localhost:3000/pin?address=${db.id}`)
-      assert.strictEqual(res.status, 200)
-      assert.strictEqual(await res.text(), `adding... ${db.id}`)
-
-      await db.add('x')
-      await db.add('y')
-      await db.add('z')
-
-      process.stdout.on('data', async data => {
-        if (data.toString().includes(`opening database from ${db.id}`)) {
-          console.log('Giving the db a few seconds to replicate...')
-
-          setTimeout(async () => {
-            console.log('...proceeding')
-            await db.drop()
-
-            const db2 = await orbitdb2.open(db.id, { sync: true })
-            assert.strictEqual(db.id, db2.id)
-
-            const res = await fetch(`http://localhost:3000/unpin?address=${db2.id}`)
-            assert.strictEqual(res.status, 200)
-            assert.strictEqual(await res.text(), `removing... ${db.id}`)
-
-            const target = 3
-            let i = 0
-            db2.events.on('replicated', (address, total) => {
-              i += total
-              if (i === target) done()
-            })
-          }, 4000)
-        }
+  async function createAndPinDb(orbitdb, name) {
+    const db = await orbitdb.log(name, { sync: true });
+    const res = await fetch(`http://localhost:3000/pin?address=${db.id}`)
+    assert.strictEqual(res.status, 200)
+    assert.strictEqual(await res.text(), `adding... ${db.id}`)
+
+    return db;
+  }
+
+  describe('Basic Functionality', function () {
+    it('Pins a database via /pin', (done) => {
+      createAndPinDb(orbitdb, 'pinTest').then(async db => {
+        await db.add('x')
+        await db.add('y')
+        await db.add('z')
+
+        process.stdout.on('data', async data => {
+          if (data.toString().includes(`opening database from ${db.id}`)) {
+            console.log('Giving the db a few seconds to replicate...')
+
+            setTimeout(async () => {
+              console.log('...proceeding')
+              await db.drop()
+
+              const db2 = await orbitdb2.open(db.id, { sync: true })
+              assert.strictEqual(db.id, db2.id)
+
+              const res = await fetch(`http://localhost:3000/unpin?address=${db2.id}`)
+              assert.strictEqual(res.status, 200)
+              assert.strictEqual(await res.text(), `removing... ${db.id}`)
+
+              const target = 3
+              let i = 0
+              db2.events.on('replicated', (address, total) => {
+                i += total
+                if (i === target) done()
+              })
+            }, 4000)
+          }
+        })
       })
     })
   })
 
+  function pause(milliseconds) {
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, milliseconds);
+    })
+  }
+
+  describe("Statistics Endpoint", function() {
+    it("Reports stats after multiple operations", async () => {
+      const db1 = await createAndPinDb(orbitdb, "statsTest0");
+      // TODO: This is troubling...
+      await pause(1000);
```

Comment:
I'm thinking on how we can do better than a 1000ms required pause between calls to `/pin`... Maybe having some ideas that I'll probably put in other PRs. 

I'm also only testing with the `eventlog` type DB and not with `docstore` or anything like that

Reply:
Gonna do this in another PR